test/widgets/message_list_test.dart

```diff
@@ -1009,4 +1009,64 @@ void main() {
       });
     });
   });
+
+  group('TypingStatus', () {
+    final zulipLocalizations = GlobalLocalizations.zulipLocalizations;
+    final users = [eg.selfUser, eg.otherUser, eg.thirdUser, eg.fourthUser];
+    final finder = find.descendant(
+      of: find.byType(TypingStatusWidget),
+      matching: find.byType(Text)
+    );
+
+    checkTyping(WidgetTester tester, TypingEvent event, {required List<User> expected}) async {
+      await store.handleEvent(event);
+      await tester.pump();
+      final otherTypist = (expected.length > 1) ? expected[1] : null;
+      check(finder.evaluate()).single.has((x) => x.widget, 'widget').isA<Text>()
+        .data.equals(zulipLocalizations.typingIndicator(
+          expected.length, expected[0].fullName, otherTypist?.fullName ?? ''));
```

Comment:
In light of my other comment, let's have these tests use concrete strings for expectations, instead of `zulipLocalizations` results :slightly_smiling_face: Try changing the tests first, and please confirm whether my prediction above is correct (at least about English).

I think just testing in English will suffice to make the point and confirm that the logic is happening outside of the i18n plural rules.